objc_interface.dd

```diff
@@ -0,0 +1,370 @@
+Ddoc
+
+$(SPEC_S Interfacing to Objective-C,
+    $(P
+        D has limited support for interfacing with Objective-C. It supports
+        external classes and calling instance methods. It is only available on
+        OS X, compiling for 64bit.
+    )
+
+    $(P
+        Fully working example is available at
+        $(LINK2 #usage-example, the bottom).
+    )
+
+    $(SECTION2 $(LNAME2 external-class, Declaring an External Class))
+
+    ---
+    extern (Objective-C)
+    interface NSObject
+    {
+        NSObject init() @selector("init");
+    }
+    ---
+
+    $(P
+        Currently all Objective-C classes needs to be declared as interfaces in
+        D. All Objective-C classes that should be accessible from within D
+        needs to be declared with the $(LINK2 #objc-linkage, Objective-C)
+        linkage.
+    )
+
+    $(P
+        The $(LINK2 #selector-attribute, @selector) attribute indicates which
+        Objective-C selector should be used when calling this method from D.
+        This attribute needs to be attached to all methods.
+    )
+
+    $(SECTION2 $(LNAME2 instance-method, Calling an Instance Method))
+
+    $(P
+        Calling an Objective-C instance method uses the same syntax as calling
+        regular D methods:
+    )
+
+    ---
+    NSObject result = object.init();
```

Comment:
This could need clarification because usually in D `init` is a property for the default initialization value. Maybe just use a more "normal" method name as an illustration to avoid unnecessary explanations at this point.